### PR TITLE
[eeprom] Add UT for eeprom_tlvinfo.py

### PR DESCRIPTION
**Pull request by keboliu**
#### Description

Add unit test cases to test  eeprom_tlvinfo.py, now code coverage is 80%

#### Motivation and Context

There is no UT available for eeprom_tlvinfo.py previously.
Using a HEX file to mock EEPROM content. Take this mocked EEPROM as input and test the functions.
Small changes to  eeprom_tlvinfo.py to improve its testability.

#### How Has This Been Tested?

```
sonic@sonic_slave_docker:/sonic/src/sonic-platform-common/tests$ pytest-3 eeprom_base_test.py
=========================================================================================== test session starts ============================================================================================
platform linux -- Python 3.9.2, pytest-6.0.2, py-1.10.0, pluggy-0.13.0 -- /usr/bin/python3
cachedir: .pytest_cache
rootdir: /sonic/src/sonic-platform-common-kebo, configfile: pytest.ini
plugins: pyfakefs-4.5.6, cov-2.10.1
collected 5 items                                                                                                                                                                                          

eeprom_base_test.py::TestEepromTlvinfo::test_eeprom_tlvinfo_read_api PASSED                                                                                                                          [ 20%]
eeprom_base_test.py::TestEepromTlvinfo::test_eeprom_tlvinfo_get_tlv_field PASSED                                                                                                                     [ 40%]
eeprom_base_test.py::TestEepromTlvinfo::test_eeprom_tlvinfo_set_eeprom PASSED                                                                                                                        [ 60%]
eeprom_base_test.py::TestEepromTlvinfo::test_eeprom_tlvinfo_update_eeprom_db PASSED                                                                                                                  [ 80%]
eeprom_base_test.py::TestEepromTlvinfo::test_eeprom_tlvinfo_read_eeprom_db PASSED   
```

### CODE REVIEW

**Comment by keboliu** [comment]
CI passed.

File: tests/eeprom_base_test.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+        """
+        if not os.path.exists(os.path.dirname(EEPROM_HEX_FILE_FULL_PATH)):
+            assert False, "File {} is not exist".format(EEPROM_HEX_FILE_FULL_PATH)
+        subprocess.check_call(['/usr/bin/xxd', '-r', '-p', EEPROM_HEX_FILE_FULL_PATH, EEPROM_SYMLINK_FULL_PATH])
+    
+    @classmethod
+    def teardown_class(cls):
+        # Remove the mock eeprom after test
+        if os.path.exists(os.path.dirname(EEPROM_HEX_FILE_FULL_PATH)):
+                subprocess.check_call(['rm', '-f', EEPROM_SYMLINK_FULL_PATH])
```
Comment (Junchao-Mellanox) [inline tests/eeprom_base_test.py]: indent issue

File: tests/eeprom_base_test.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        eeprom_new = eeprom_class.set_eeprom(eeprom, ['0x2E = A2'])
+        (is_valid, t) = eeprom_class.get_tlv_field(eeprom_new, 0x2E)
+        assert(is_valid and t[2].decode("ascii").rstrip('\0') == 'A2')
+
+        # Test adding invalid field
+        (is_valid, t) = eeprom_class.get_tlv_field(eeprom_new, 0x20)
+        assert(not is_valid)
+        with mock.patch('sys.exit') as exit_mock:
+            eeprom_new = eeprom_class.set_eeprom(eeprom, ['0x20 = Invalid'])
+        assert exit_mock.called
```
Comment (Junchao-Mellanox) [inline tests/eeprom_base_test.py]: should the assert be inside "with"?

File: tests/eeprom_base_test.py [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+    def test_eeprom_tlvinfo_update_eeprom_db(self):
+        # Test updating eeprom to DB by mocking redis hmset
+        eeprom_class = eeprom_tlvinfo.TlvInfoDecoder(EEPROM_SYMLINK_FULL_PATH, 0, '', True)
+        eeprom = eeprom_class.read_eeprom()
+        eeprom_class.redis_client.hmset = mock.MagicMock(return_value = True)
+        assert(0 == eeprom_class.update_eeprom_db(eeprom))
```
Comment (Junchao-Mellanox) [inline tests/eeprom_base_test.py:163]: should the db content be checked?

File: tests/eeprom_base_test.py (hunk 1 above)
Comment (keboliu) [reply tests/eeprom_base_test.py]: fixed

File: tests/eeprom_base_test.py (hunk 2 above)
Comment (keboliu) [reply tests/eeprom_base_test.py]: fixed

File: tests/eeprom_base_test.py (hunk 3 above)
Comment (keboliu) [reply tests/eeprom_base_test.py:163]: the framework not support redis-db yet, here just check the logic before set to DB.

**Review by Junchao-Mellanox — APPROVED** [review]

**Review by stephenxs — APPROVED** [review]

**Pull request closed without merging**